### PR TITLE
fix:added code refactoring changes for 2378

### PR DESCRIPTION
**Pull request by Ap284**
Pull request title

- Clearly and concisely describes what it does
- Refer to the issue that it solves, if available

Pull request description

- Describes the main changes that come with the pull request
- Any relevant additional information is provided

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

File: collection-pipeline/pom.xml [hunk 1]
```diff
@@ -57,6 +57,14 @@
...
         </executions>
       </plugin>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+                <source>16</source>
+                <target>16</target>
+            </configuration>
+        </plugin>
```
Comment (iluwatar) [inline collection-pipeline/pom.xml:67]: This is now set in the parent pom.xml. Please rebase your changes on top of `master`. Let's see if we get a passing build.

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: collection-pipeline/pom.xml (hunk 1 above)
Comment (iluwatar) [reply collection-pipeline/pom.xml]: You have the same configuration elsewhere too. Please check throughout the code.

**Comment by iluwatar** [comment]
Please try to address the failing build

**Review by iluwatar — CHANGES_REQUESTED** [review]
The build needs to pass before we can merge

**Comment by Ap284** [comment]
Hi @iluwatar,
I have rebased changes with master still getting same issue:

Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.8.0.2131:sonar (default-cli) on project java-design-patterns: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator -> [Help 1]

**Comment by Ap284** [comment]
Hi @iluwatar,

I have sync the fork with the upstream master branch. But, still getting the same issue:

Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.8.0.2131:sonar (default-cli) on project java-design-patterns: Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator -> [Help 1]

Please, guide me if I missed anything or anything incorrect.

**Comment by iluwatar** [comment]
The build works for me locally, so I don't think there's anything wrong in this branch. I recently reconfigured the Sonar scan, but maybe it's using the old one from the cache or something like that. Anyway, I'm going to merge this.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
@all-contributors please add @Ap284 for code

**Pull request merged**